tests/pytests/test_issues.py

```diff
@@ -857,46 +857,54 @@ def mod5778_add_new_shard_to_cluster(env):
     new_instance = subprocess.Popen(cmd_args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
 
     # Connect the new instance to the cluster (making sure the new instance didn't crash)
-    env.cmd('CLUSTER', 'MEET', '127.0.0.1', new_instance_port)
-    time.sleep(5)
+    env.expect('CLUSTER', 'MEET', '127.0.0.1', new_instance_port).ok()
+    kwargs = {'host': '127.0.0.1', 'port': new_instance_port, 'decode_responses': True, 'password': env.envRunner.password}
     if env.envRunner.isTLS():
-        new_instance_conn = RedisCluster(host='127.0.0.1', port=new_instance_port, decode_responses=True,
-                                         ssl=True,
-                                         ssl_keyfile=env.envRunner.shards[0].getTLSKeyFile(),
-                                         ssl_certfile=env.envRunner.shards[0].getTLSCertFile(),
-                                         ssl_cert_reqs=None,
-                                         ssl_ca_certs=env.envRunner.shards[0].getTLSCACertFile(),
-                                         ssl_password=env.envRunner.tlsPassphrase,
-                                         password=env.envRunner.password)
-    else:
-        new_instance_conn = RedisCluster(host='127.0.0.1', port=new_instance_port, decode_responses=True,
-                                         password=env.envRunner.password)
-    env.assertEqual(new_instance_conn.ping(), True)
+      kwargs.update({'ssl': True,
+                     'ssl_keyfile': env.envRunner.shards[0].getTLSKeyFile(),
+                     'ssl_certfile': env.envRunner.shards[0].getTLSCertFile(),
+                     'ssl_cert_reqs': None,
+                     'ssl_ca_certs': env.envRunner.shards[0].getTLSCACertFile(),
+                     'ssl_password': env.envRunner.tlsPassphrase})
+    with TimeLimit(10, 'waiting for new shard to acknowledge the topology change'):
+      while True:
+        time.sleep(0.05)
+        try:
+          new_instance_conn = RedisCluster(**kwargs)
+          break
+        except:
+          pass
```

Comment:
Did you understand why did it take longer for the new shard to upload after your changes? Or it is not related?

Reply:
No. Here it was just flaky, replacing arbitrary `sleep(5)` with a timed-out check.
Why was it taking 5 seconds before?